src/migration/1712315208717-PostSlugMaxLength.ts

```diff
@@ -0,0 +1,72 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class PostSlugMaxLength1712315208717 implements MigrationInterface {
+  name = 'PostSlugMaxLength1712315208717';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "typeorm_metadata" REPLICA IDENTITY FULL`,
+    );
+    await queryRunner.query(`DROP INDEX "public"."IDX_post_slug"`);
+    await queryRunner.query(
+      `ALTER TABLE "post" DROP CONSTRAINT "UQ_cd1bddce36edc3e766798eab376"`,
+    );
+    await queryRunner.query(`ALTER TABLE "post" DROP COLUMN "slug"`);
+    await queryRunner.query(
+      `DELETE FROM "public"."typeorm_metadata" WHERE "type" = $1 AND "name" = $2 AND "database" = $3 AND "schema" = $4 AND "table" = $5`,
+      ['GENERATED_COLUMN', 'slug', 'api', 'public', 'post'],
+    );
+    await queryRunner.query(
+      `ALTER TABLE "post" ADD "slug" text GENERATED ALWAYS AS (trim(BOTH '-' FROM regexp_replace(lower(trim(COALESCE(LEFT(post.title,100),'')||'-'||post.id)), '[^a-z0-9-]+', '-', 'gi'))) STORED NOT NULL`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "post" ADD CONSTRAINT "UQ_cd1bddce36edc3e766798eab376" UNIQUE ("slug")`,
+    );
+    await queryRunner.query(
+      `INSERT INTO "public"."typeorm_metadata"("database", "schema", "table", "type", "name", "value") VALUES ($1, $2, $3, $4, $5, $6)`,
+      [
+        'api',
+        'public',
+        'post',
+        'GENERATED_COLUMN',
+        'slug',
+        "trim(BOTH '-' FROM regexp_replace(lower(trim(COALESCE(LEFT(post.title,100),'')||'-'||post.id)), '[^a-z0-9-]+', '-', 'gi'))",
+      ],
+    );
+    await queryRunner.query(
+      `CREATE UNIQUE INDEX "IDX_post_slug" ON "post" ("slug") `,
+    );
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`DROP INDEX "public"."IDX_post_slug"`);
+    await queryRunner.query(
+      `DELETE FROM "public"."typeorm_metadata" WHERE "type" = $1 AND "name" = $2 AND "database" = $3 AND "schema" = $4 AND "table" = $5`,
+      ['GENERATED_COLUMN', 'slug', 'api', 'public', 'post'],
+    );
+    await queryRunner.query(
+      `ALTER TABLE "post" DROP CONSTRAINT "UQ_cd1bddce36edc3e766798eab376"`,
+    );
+    await queryRunner.query(`ALTER TABLE "post" DROP COLUMN "slug"`);
+    await queryRunner.query(
```

Comment:
I know I said generated columns but seeing this it might be better long term to just have it as a regular column to avoid dropping each time we change something and instead do it with migrations? Not blocking for this though.

Reply:
We can still have it be readonly from API/ORM side.

Reply:
Hmm hoping we don't have to change often?
It's kind of the same as trigger issues.

For now let's leave it, let's keep in mind if there's another issue, we go for text and just modify the value?

Reply:
Agreed, non blocking for this 👍